src/index.js

```diff
@@ -279,15 +279,24 @@ function sanitizeHtml(html, options, _recursing) {
                 // naughtyHref is in charge of whether protocol relative URLs
                 // are cool. We should just accept them
                 parsed = url.parse(value, false, true);
+                parsed.domain = parsed.hostname && parsed.hostname.substring(
+                  parsed.hostname.lastIndexOf('.', parsed.hostname.lastIndexOf('.') - 1) + 1
```

Comment:
If example.com is an allowed domain, this approach will match foo.example.com, but not foo.2.example.com.

A better approach would be to check whether the hostname either exactly matches the domain, or ends in a period followed by the domain. Then the number of dots in the prefix is not relevant.

Reply:
It really works.
![image](https://user-images.githubusercontent.com/7382178/83325787-c9e11200-a277-11ea-982a-6e9149b29cf8.png)
But I agree that it isn't a clear piece of code and it is better to use `endsWith` checking.